pandas/io/tests/parser/test_network.py

```diff
@@ -7,31 +7,48 @@
 
 import os
 import nose
+import functools
 
 import pandas.util.testing as tm
 from pandas import DataFrame
 from pandas import compat
 from pandas.io.parsers import read_csv, read_table
 
 
-class TestUrlGz(tm.TestCase):
-
-    def setUp(self):
-        dirpath = tm.get_data_path()
-        localtable = os.path.join(dirpath, 'salaries.csv')
-        self.local_table = read_table(localtable)
-
-    @tm.network
-    def test_url_gz(self):
-        url = ('https://raw.github.com/pandas-dev/pandas/'
-               'master/pandas/io/tests/parser/data/salaries.csv.gz')
-        url_table = read_table(url, compression="gzip", engine="python")
-        tm.assert_frame_equal(url_table, self.local_table)
-
-    @tm.network
-    def test_url_gz_infer(self):
-        url = 'https://s3.amazonaws.com/pandas-test/salary.table.gz'
-        url_table = read_table(url, compression="infer", engine="python")
+class TestCompressedUrl(object):
+
+    compression_to_extension = {
+        'gzip': '.gz',
+        'bz2': '.bz2',
+        'zip': '.zip',
+        'xz': '.xz',
+    }
+
+    def __init__(self):
+        path = os.path.join(tm.get_data_path(), 'salaries.csv')
+        self.local_table = read_table(path)
+        self.base_url = ('https://github.com/pandas-dev/pandas/raw/master/'
+                         'pandas/io/tests/parser/data/salaries.csv')
+
+    def test_compressed_urls(self):
+        """Test reading compressed tables from URL."""
+        msg = ('Test reading {}-compressed tables from URL: '
+               'compression="{}", engine="{}"')
+
+        for compression, extension in self.compression_to_extension.items():
+            url = self.base_url + extension
+            # args is a (compression, engine) tuple
+            for args in [(compression, 'python'), ('infer', 'python')]:
```

Comment:
Not sure about that. Let's see what the test runs says, but if there are problems with the C engine, let's leave that for another PR. 
In any case, if those pass for the C engine, you can also use `self.read_csv` instead of `read_csv` and then the tests will automatically be run for both the python and C engine without needing to specify it explicitly